### PR TITLE
Upgrade SocketRocket to 0.7.0

### PR DESCRIPTION
**Pull request by amgleitman**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [ ] I am making a fix / change for the macOS implementation of react-native
- [x] I am making a change required for Microsoft usage of react-native

## Summary

This upgrades the SocketRocket to 0.7.0, which fixes a number of retain cycle issues.

## Changelog

[GENERAL] [FIXED] - Fix retain cycle issues in SocketRocket

## Test Plan

Validated that the WebSocket test page still works.
If the CIs also pass, we should be good.

### CODE REVIEW

File: React-Core.podspec [hunk 1]
```diff
@@ -18,7 +18,7 @@ end
 
 folly_compiler_flags = '-DFOLLY_NO_CONFIG -DFOLLY_MOBILE=1 -DFOLLY_USE_LIBCPP=1 -Wno-comma -Wno-shorten-64-to-32'
 folly_version = '2021.07.22.00'
-socket_rocket_version = '0.6.0'
+socket_rocket_version = '0.7.0'
```
Comment (Saadnajmi) [inline React-Core.podspec:21]: macOS tags?

**Review by Saadnajmi — APPROVED** [review]
Can we also port to 0.71-stable and maybe make an upstream PR?

File: React-Core.podspec (hunk 1 above)
Comment (amgleitman) [reply React-Core.podspec]: Added with 7f3eb5b.

**Comment by amgleitman** [comment]
> Can we also port to 0.71-stable and maybe make an upstream PR?

0.71-stable port is #1888, upstream PR will happen soon as well.

**Pull request merged**